Animations/Animation.js

```diff
@@ -37,20 +22,36 @@ var makeTransition = function (displayName, transitions) {
       // componentDidEnter FUNCTION: use to define actions to run after an animation has entry completed
       // componentDidLeave FUNCTION: use to define actions to run after an animation has leave completed
   return React.createClass({
+    // setAnimationDomNode FUNCTION: function to handle manipulating and setting a TransitionGroup in the DOM
+      //@param action STRING: the transition key you are looking to affect
+      //@param callback FUNCTION: callback attribute passed in from the containing function to act on function completion
+    setAnimationDomNode: function (action, callback) {
+      this.el = this.getDOMNode();
+      this.$el = $(this.el);
+
+      // requestAnimationFrame FUNCTION: Calls the specified function updating an animation before the next browser repaint. Defined in window
+      requestAnimationFrame(function () {
+        this.$el.css(transitions[action]);
+        requestAnimationFrame(function () {
+          this.$el.css(transitions[action + 'Active']);
+          Arrival(this.$el, callback);
+        }.bind(this));
+      }.bind(this));
+    },
     componentWillEnter: function (callback) {
-      setAnimationDomNode('enter', callback);
+      this.setAnimationDomNode('enter', callback);
     },
 
     componentDidEnter: function () {
```

Comment:
I would prefer to have it left in with the addition of 'this.setAnimationDomNode('entered', callback)'. Provided the callback is something that is passed in and should be passed into setAnimationDomNode.